src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -1002,6 +1112,105 @@ internal static char ParseFormatSpecifier(ReadOnlySpan<char> format, out int dig
             }
         }
 
+        private static string? FormatBigIntegerToBin(bool targetSpan, BigInteger value, int digits, Span<char> destination, out int charsWritten, out bool spanSuccess)
+        {
+            // Get the bytes that make up the BigInteger.
+            byte[]? arrayToReturnToPool = null;
+            Span<byte> bytes = stackalloc byte[64]; // arbitrary threshold
```

Comment:
We can make this larger, like 256. Do we use 64 as such an arbitrary threshold in other BigInteger formatting routines?

Reply:
Ah, nevermind, I see you might be stackalloc'ing 1K of chars below.